### DISCUSSION TITLE
Delete complete Tenant including dependent objects

### DISCUSSION THREAD

**Discussion by SebastiaanOsinga** [netbox-community/netbox#9805] [Ideas]
It would be nice to delete complete Tenants including their dependencies.
I understand that some dependencies could be used for multiple tenants or sites, but it could be unlinked from the tenant which you want to delete.

if not possible, or too complex to perform, perhaps it would be nice, if you could export tenants, including all depended object, so you can start up a new instance of netbox and import it there.

I'm current running netbox v3.0.9 ( planning to upgrade ) and have multiple tenants. When a Tenant leaves, we have to delete all data. It would be very helpful if this is possible. 

if all the above is not possible. can someone have a SQL query to make this happen?

**Comment by candlerb** [comment]
How about writing a [Custom Script](https://docs.netbox.dev/en/stable/customization/custom-scripts/) which does this?  All it needs to do is to prompt for a tenant using ObjectVar and then:

```
    IPAddress.objects.filter(tenant=tenant).delete()
    Device.objects.filter(tenant=tenant).delete()
    VirtualMachine.objects.filter(tenant=tenant).delete()
    ... etc
    Tenant.objects.filter(tenant=tenant).delete()
```

See https://docs.djangoproject.com/en/4.0/ref/models/querysets/#delete

However, personally I don't think this is what I'd for. For example, if a tenant leaves, all their devices don't magically vanish from the racks.  Either you need a process to remove all those devices (in which case you might set the status to "Decommissioning" and then only remove them from Netbox after they've been physically removed); or the device remains but no longer assigned to that tenant (in which case you want to set the tenant to null, not delete the device)
